### ISSUE TITLE
Filter `name` when using choices with name/value

### ISSUE THREAD

**Issue by sindresorhus** [SBoudrias/Inquirer.js#150]
Here only the value is returned, but I want to also be able to change the name, as I want it to be show differently in the CLI.

![2014-08-16 at 00 10](https://cloud.githubusercontent.com/assets/170270/3939550/17246766-24c9-11e4-9eb5-78203c18ff84.png)

Should be able to change this text.

```js
var inquirer = require("../lib/inquirer");

inquirer.prompt([
  {
    type: "list",
    name: "theme",
    message: "What do you want to do?",
    choices: [
      {value: 'foo1', name: "Order a pizza"},
      {value: 'foo2', name: "Make a reservation"},
      {value: 'foo3', name: "Ask opening hours"},
      {value: 'foo4', name: "Talk to the receptionnist"}
    ],
    filter: function( val ) {
      return val[0];
    }
  }
], function( answers ) {
    console.log( JSON.stringify(answers, null, "  ") );
  });
```

Maybe return an object in the `filter` function instead which would enable you to change both.

**Comment by dantman** [comment]
I'm not sure what the point of doing this as a filter would be. I'd like this too, but I think it should be an optional property on the choice object.

```js
choices: [
  { value: 'pizza', name: "Order a pizza", short: "Pizza" }
]
```

Or maybe a `pretty` or other property to override the text shown in the list instead.

I do think this is fairly important, because there are various cli apps where the name is made really long and formatted so it's helpful to the user. But all that formatting is in the way after the selection.

For example `npm-check -u` which turns into this after selection.
```
? Choose which packages to update. bluebird  2.10.0  ❯  2.10.1  https://github.com/petkaantonov/bluebird, req...
```
This got cut off because it line-wrapped and ended up written over by the continuation of npm-check.
((This is actually cut off

**Comment by SBoudrias** [comment]
@dantman Ah, that would be very simple to implement and it would probably resolve most use case for an answer filtering option.

How do you feel about this @sindresorhus?

**Comment by SBoudrias** [comment]
Example implementation here: https://github.com/SBoudrias/Inquirer.js/tree/short-name-for-choices

**Comment by ascariandrea** [comment]
@SBoudrias any update of this?

**Comment by SBoudrias** [comment]
@ascariandrea I was waiting for feedback on the suggested API.

**Comment by ascariandrea** [comment]
I'd suggest to make the list you are filtering on visible and make it works like `rawlist`.

**Comment by sindresorhus** [comment]
Looks like this was fixed in 0.11.0: https://github.com/SBoudrias/Inquirer.js/releases/tag/v0.11.0
